test/ExportVerilog/rtl-dialect.mlir

```diff
@@ -65,27 +69,35 @@ module {
-               %28, %29, %30, %31, %33, %34, %35, %36, %37 :
+               %28, %29, %30, %31, %33, %34, %35, %36, %37, %39 :
      i4,i4, i4,i4,i4,i4,i4, i4,i4,i4,i4,i4,
      i4,i1,i1,i1,i1, i1,i1,i1,i1,i1, i1,i1,i1,i1,
-     i12, i2,i9,i4, i4, !rtl.array<3xi4>, i12, i4
+     i12, i2,i9,i4, i4, !rtl.array<3xi4>, i12, i4, !rtl.struct<foo: i2, bar: i3>
   }
   // CHECK-LABEL: module TESTSIMPLE(
-  // CHECK-NEXT:   input  [3:0]            a, b
-  // CHECK-NEXT:   input                   cond,
-  // CHECK-NEXT:   input  [11:0][9:0][3:0] array2d,
-  // CHECK-NEXT:   input  [7:0]            uarray[15:0],
-  // CHECK-NEXT:   output [3:0]            r0, r2, r4, r6, r7, r8, r9, r10, r11, r12, r13, r14, r15
-  // CHECK-NEXT:   output                  r16, r17, r18, r19, r20, r21, r22, r23, r24, r25, r26, r27, r28
-  // CHECK-NEXT:   output [11:0]           r29,
-  // CHECK-NEXT:   output [1:0]            r30,
-  // CHECK-NEXT:   output [8:0]            r31,
-  // CHECK-NEXT:   output [3:0]            r33, r34,
-  // CHECK-NEXT:   output [2:0][3:0]       r35,
-  // CHECK-NEXT:   output [11:0]           r36,
-  // CHECK-NEXT:   output [3:0]            r37);
+  // CHECK-NEXT:   input  [3:0]                                             a, b,
+  // CHECK-NEXT:   input                                                    cond,
+  // CHECK-NEXT:   input  [9:0][3:0]                                        array1,
+  // CHECK-NEXT:   input  [7:0]                                             uarray1[15:0], uarray2[15:0],
+  // CHECK-NEXT:   input  struct packed {logic [1:0] foo; logic [2:0] bar;} structA,
+  // CHECK-NEXT:   input  [11:0][9:0][3:0]                                  array2d,
+  // CHECK-NEXT:   input  struct packed {logic [7:0] baz;} [1:0]            aofs,
+  // CHECK-NEXT:   output [3:0]                                             r0, r2, r4, r6, r7, r8, r9, r10, r11, r12, r13, r14, r15,
+  // CHECK-NEXT:   output                                                   r16, r17, r18, r19, r20, r21, r22, r23, r24, r25, r26, r27,
+  // CHECK-NEXT:   output                                                   r28,
+  // CHECK-NEXT:   output [11:0]                                            r29,
+  // CHECK-NEXT:   output [1:0]                                             r30,
+  // CHECK-NEXT:   output [8:0]                                             r31,
+  // CHECK-NEXT:   output [3:0]                                             r33, r34,
+  // CHECK-NEXT:   output [2:0][3:0]                                        r35,
+  // CHECK-NEXT:   output [11:0]                                            r36,
+  // CHECK-NEXT:   output [3:0]                                             r37,
+  // CHECK-NEXT:   output struct packed {logic [1:0] foo; logic [2:0] bar;} r38);
   // CHECK-EMPTY:
   // CHECK-NEXT:   wire [8:0][3:0] [[WIRE0:.+]] = {{[{}][{}]}}4'hF}, {4'hF}, {4'hF}, {4'hF}, {4'hF}, {4'hF}, {4'hF}, {4'hF}, {4'hF}};
```

Comment:
I would also suggest a test for a struct declaration. It doesn't _look_ like this is supported by the code in this PR. Are inline structs in declarations not well supported?

Reply:
added for the simple case, and filed an issue for the general case: https://github.com/llvm/circt/issues/530